vignettes/IRangesOverview.Rmd

```diff
@@ -0,0 +1,527 @@
+
+### Making ranges disjoint
+
+A more complex type of operation is making a set of ranges disjoint, *i.e.*
+non-overlapping. For example, `threebands` returns a disjoint set of three
+ranges for each input range.
+
+The `disjoin` function makes an *IRanges* object disjoint by fragmenting it into
+the widest ranges where the set of overlapping ranges is the same.
+
+<a id="fig-ranges-disjoin"></a>
+```{r ranges-disjoin, fig.cap = "Plot of disjoined ranges.", fig.height=2.25}
+disjoin(ir)
+plotRanges(disjoin(ir))
+```
+
+A variant of `disjoin` is `disjointBins`, which divides the ranges into bins,
+such that the ranges in each bin are disjoint. The return value is an integer
+vector of the bins.
+
+```{r ranges-disjointBins}
+disjointBins(ir)
+```
+
+### Other transformations
+
+Other transformations include `reflect` and `flank`. The former "flips" each
+range within a set of common reference bounds.
+
+```{r ranges-reflect}
+reflect(ir, IRanges(start(ir), width=width(ir)*2))
+```
+
+The `flank` returns ranges of a specified width that flank, to the left
+(default) or right, each input range. One use case of this is forming promoter
+regions for a set of genes.
+
+```{r ranges-flank}
+flank(ir, width=seq_len(length(ir)))
+```
+
+## Set Operations
+
+Sometimes, it is useful to consider an *IRanges* object as a set of integers,
+although there is always an implicit ordering. This is formalized by
+*NormalIRanges*, above, and we now present versions of the traditional
+mathematical set operations *complement*, *union*, *intersect*, and *difference*
+for *IRanges* objects. There are two variants for each operation. The first
+treats each *IRanges* object as a set and returns a *normal* value, while the
+other has a "parallel" semantic like `pmin`/`pmax` and performs the operation
+for each range pairing separately.
+
+The *complement* operation is implemented by the `gaps` and `pgap` functions. By
+default, `gaps` will return the ranges that fall between the ranges in the
+(normalized) input. It is possible to specify a set of bounds, so that flanking
+ranges are included.
+
+<a id="fig-ranges-gap"></a>
+```{r ranges-gaps, fig.cap = "Plot of gaps from ranges.", fig.height=2.25}
+gaps(ir, start=1, end=50)
+plotRanges(gaps(ir, start=1, end=50), c(1,50))
+```
+
+`pgap` considers each parallel pairing between two *IRanges* objects and finds
+the range, if any, between them. Note that the function name is singular,
+suggesting that only one range is returned per range in the input.
+
+```{r ranges-pgap}
+```
+
+The remaining operations, *union*, *intersect* and *difference* are implemented
+by the `[p]union`, `[p]intersect` and `[p]setdiff` functions, respectively.
+These are relatively self-explanatory.
+
+```{r ranges-union}
+```
+
+```{r ranges-punion}
+```
+
+```{r ranges-intersect}
+```
+
+```{r ranges-pintersect}
+```
+
+```{r ranges-setdiff}
+```
+
+```{r ranges-psetdiff}
+```
+
+<!-- \subsection{Mapping Ranges Between Vectors} --> 
```

Comment:
@hpages This is just a suggestion, but maybe this section can be removed? 

```
```{r ranges-union}
```

```{r ranges-punion}
```

```{r ranges-intersect}
```

```{r ranges-pintersect}
```

```{r ranges-setdiff}
```

```{r ranges-psetdiff}
```

<!-- \subsection{Mapping Ranges Between Vectors} -->
```

Reply:
I have not yet removed it. I will remove it after receiving confirmation from @hpages.

Reply:
Sure, please remove. These are placeholders that were intended to be filled with something at some point but it never happened. Thx!

Reply:
Done